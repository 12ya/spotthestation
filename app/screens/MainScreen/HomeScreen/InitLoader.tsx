import React from "react"
import { ViewStyle, View, TextStyle } from "react-native"
import { Text } from "../../../components"
import { fontSizes, lineHeights, scale, typography } from "../../../theme"
import { colors } from "../../../theme/colors"
import { normalizeHeight } from "../../../utils/normalizeHeight"

export function InitLoader() {
  return (
    <View 
      accessible
      accessibilityLabel="coach mark"
      accessibilityHint="coach mark"
      accessibilityRole="text"
      style={$modalBodyContainer}
    >
      <Text tx="homeScreen.initLoader.message" style={$title} />
    </View>
  )
}

const $modalBodyContainer: ViewStyle = {
  backgroundColor: colors.palette.buttonBlue,
  borderRadius: scale(16),
  alignItems: 'center',
  paddingVertical: 36,
  paddingHorizontal: 30,
  width: '100%',
  alignSelf: 'center',
  marginTop: normalizeHeight(.28)
}

const $title: TextStyle = {
  fontFamily: typography.primary.normal,
  fontSize: fontSizes[24],
  lineHeight: lineHeights[29],
  color: colors.palette.neutral100,
  paddingBottom: 12,
  paddingTop: 18
}
