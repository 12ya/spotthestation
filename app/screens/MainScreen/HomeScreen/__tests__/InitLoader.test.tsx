import React from "react"
import { InitLoader } from "../InitLoader"
import { render } from "@testing-library/react-native"

it("renders correctly", () => {
  const tree = render(<InitLoader />).toJSON()
  expect(tree).toMatchSnapshot()
})
