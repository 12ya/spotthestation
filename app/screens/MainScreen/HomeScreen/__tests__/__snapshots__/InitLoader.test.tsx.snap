// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityHint="coach mark"
  accessibilityLabel="coach mark"
  accessibilityRole="text"
  accessible={true}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#0B3D91",
      "borderRadius": 23,
      "marginTop": 373.52000000000004,
      "paddingHorizontal": 30,
      "paddingVertical": 36,
      "width": "100%",
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {},
        Array [
          Object {
            "fontSize": 23,
            "lineHeight": 35,
          },
          Object {
            "fontFamily": "Helvetica Neue",
          },
          Object {
            "color": "#191015",
          },
        ],
        undefined,
        undefined,
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Helvetica Neue",
          "fontSize": 35,
          "lineHeight": 42,
          "paddingBottom": 12,
          "paddingTop": 18,
        },
      ]
    }
  >
    homeScreen.initLoader.message undefined
  </Text>
</View>
`;
